**Naszym celem w tym projekcie jest znalezienie profili aplikacji mobilnych, które będą opłacalne dla rynków App Store i Google Play. Pracujemy jako analitycy danych w firmie tworzącej aplikacje mobilne na Androida i iOS,       a naszym zadaniem jest umożliwienie naszemu zespołowi programistów podejmowania decyzji opartych na danych w odniesieniu do rodzaju tworzonych przez nich aplikacji.**

W naszej firmie tworzymy wyłącznie aplikacje, które można pobrać i zainstalować bezpłatnie, a naszym głównym źródłem przychodów są reklamy w aplikacjach. Oznacza to, że na nasze przychody z dowolnej aplikacji największy wpływ ma liczba użytkowników korzystających z naszej aplikacji. Naszym celem w tym projekcie jest analiza danych, aby pomóc naszym programistom zrozumieć, jakie rodzaje aplikacji mogą przyciągnąć więcej użytkowników.

Zbiór danych zawierający dane dotyczące około dziesięciu tysięcy aplikacji na Androida z Google Play. Zestaw danych możesz pobrać bezpośrednio pod tym linkiem.
Zbiór danych zawierający dane dotyczące około siedmiu tysięcy aplikacji na iOS z App Store. Zestaw danych możesz pobrać bezpośrednio pod tym 
[linkiem](https://dq-content.s3.amazonaws.com/350/googleplaystore.csv).

In [1]:
from csv import reader

opened_file = open('googleplaystore.csv')
read_file = reader(opened_file)
android = list(read_file)
android_header = android[0]
android = android[1:]


opened_file = open('AppleStore.csv')
read_file = reader(opened_file)
ios = list(read_file)
ios_header = ios[0]
ios = ios[1:]

In [2]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') 
        
    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

print(android_header)
print('\n')
explore_data(android, 0, 3, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


In [3]:
print(ios_header)
print('\n')
explore_data(ios, 0, 3, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 7197
Number of columns: 16


In [4]:
#Drukowanie błędnego wiersza
print(android[10472]) 
print('\n')
print(android_header) 
print('\n')
print(android[0])   
#Usuwanie błędnego wiersza
print(len(android))
del android[10472]
print(len(android))

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']
10841
10840


In [5]:
#Zbiór danych Google Play zawiera zduplikowane wpisy:
for app in android:
    name = app[0]
    if name == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


In [6]:
#Zliczanie liczby duplikatów:
duplicate_apps = []
unique_apps = []
for app in android:
    name= app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)
print("Number of duplicate apps:" , len(duplicate_apps))
print("\n")
print("Examples of duplicate apps: " , duplicate_apps[:15])

Number of duplicate apps: 1181


Examples of duplicate apps:  ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


*Nie chcę liczyć aplikacji kilka razy, dlatego usuwam zduplikowane wpisy. Mogłabym usuwać je losowo jednak moim kryterium będzie ilość komentarzy. Zostawiam aplikacje, w których jest ich najwięcej.*


In [7]:
#Tworzenie słownika, w którym każdy klucz jest unikalną nazwą aplikacji, a odpowiadająca mu wartość słownika to największa liczba recenzji tej aplikacji.

reviews_max = {}

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
        
    elif name not in reviews_max:
        reviews_max[name] = n_reviews

In [8]:
#Obecne dane- długość słownika
print("Expected length: " , len(android) - 1181)

Expected length:  9659


In [9]:
#Używam słownika utworzonego powyżej do usunięcia zduplikowanych wierszy:
android_clean = []
already_added = []

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if (reviews_max[name] == n_reviews) and (name not in already_added):
        android_clean.append(app)
        already_added.append(name)

In [10]:
#Sprawdzenie czy dobrze usunełam (czy zbiór ten ma 9659 wierszy):
explore_data(android_clean, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9659
Number of columns: 13


In [11]:
print(ios[813][1])
print(ios[6731][1])
print("\n")
print(android_clean[4412][0])
print(android_clean[7940][0])

爱奇艺PPS -《欢乐颂2》电视剧热播
【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜


中国語 AQリスニング
لعبة تقدر تربح DZ


*W analizowanych danych znajdują się aplikacje w innym języku niż angielski. Mogę je usunąć wiedząc, iż każdej literze w języku angielskim przypada liczba mniejsza niż 127 zgodnie z systemem American Standard Code for Information Interchange. Dzięki czemu usuwając wszystkie liczby od 127 w górę, będę posiadać zbiór tylko anglojęzycznych aplikacji, ponieważ ich nazwy składają się z liter używanych w tym języku.*



Najpierw jednak tworzę funkcję, która pobiera ciąg znaków i zwraca, Falsejeśli w ciągu znajduje się jakikolwiek znak, który nie należy do zbioru popularnych znaków angielskich, w przeciwnym razie funkcja zwraca True.

Ten kod jes w markdown, poneważ nie jest do końca poprawny:

def is_english(string):
    
    for character in string:
        if ord(character) > 127:
            return False
    
    return True

#Sprawdzam, czy funkcja działa poprawnie:

print(is_english('Instagram'))

print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))

print(is_english('Docs To Go™ Free Office Suite'))

print(is_english('Instachat 😜'))




*Niestyety powyższa funkcja nie dział poprawnie, ponieważ w nazwach aplikacji mamy emotikony oraz znaki takie jak TM, które mają liczby powyżej 127 w systemie ASCII przez co aplikacje posiadające je są brane za nieanglojęzyczne pomimo tego, że są napisane w języku angielskim. Dlatego by zminimalizować wpływ utraty danych, usunę aplikację tylko wtedy, gdy jej nazwa zawiera więcej niż trzy znaki i odpowiadające im liczby spoza zakresu ASCII. Oznacza to, że wszystkie aplikacje w języku angielskim zawierające maksymalnie trzy emoji lub inne znaki specjalne będą nadal oznaczone jako angielskie.*

In [12]:
#Poprawnie działająca funkcja (zwraca True, jeśli nazwa aplikacji jest w języku angielskim):
    
def is_english(string):
    non_ascii = 0
    
    for character in string:
        if ord(character) > 127:
            non_ascii += 1
    
    if non_ascii > 3:
        return False
    else:
        return True

print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

True
True


In [13]:
#Filtruje aplikacje w języku innym niż angielski z obu zbiorów danych. Przeglądam zestaw danych.
android_english = []
ios_english = []

for app in android_clean:
    name = app[0]
    if is_english(name):
        android_english.append(app)
        
for app in ios:
    name = app[1]
    if is_english(name):
        ios_english.append(app)
        
explore_data(android_english, 0, 3, True)
print('\n')
explore_data(ios_english, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 

In [14]:
#Wizolowanie bezpłatnych aplikacji:

android_final = []
ios_final = []

for app in android_english:
    price = app[7]
    if price == '0':
        android_final.append(app)
        
for app in ios_english:
    price = app[4]
    if price == '0.0':
        ios_final.append(app)
        
print(len(android_final))
print(len(ios_final))

8864
3222


Moim zadnaniem jest określenie rodzajów aplikacji, które prawdopodbnie przyciągną więcej użytkowników, ponieważ liczba osób korzystająca z danej aplikacji zwiększa jej przychody. 

Aby zminimalizować ryzyko i koszty ogólne strategia dalidacji pomysłu na aplikację składa się z trzech etapów:
1. Utwórzenia minimalnej wersji aplikacji na Androida i dodanie jej do Google Play.
2. Jeśli aplikacja cieszy się dobrą reakcją użytkowników, rozwijam ją dalej.
3. Jeśli aplikacja zacznie przynosić zyski po sześciu miesiącach, buduję wersję aplikacji na iOS i dodaję ją do App Store.

Moim ostatecznym celem jest dodanie aplikacji zarówno do Google Play jak i Apple Store, dlatego muszę znaleść aplikacje, które odniosą sukces na obu rynkch.


In [15]:
#Tworzenie funkcji freq_table(), która zwraca tabelę częstości dla dowolnej kolumny (częstość jest wyrażona w %):

def freq_table(dataset, index):
    table = {}
    total = 0
    
    for row in dataset:
        total += 1
        value = row[index]
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
    
    table_percentages = {}
    for key in table:
        percentage = (table[key] / total) * 100
        table_percentages[key] = percentage 
    
    return table_percentages

#Tworzę funkcję display_table aby wyświetlić tabelę częstości kolumn rime_genre, Genresi Category:

def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
        
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

In [16]:
#Analiza tebeli częstości wygenerowanej dla prime_genre:
display_table(ios_final, -5)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


*Na podstawie powyższych danych możemy wyciągnąć wnioski, że najczęstrzym i przeważającym nad innymi gatunkiem aplikacji są gry. Większość aplikacji jest stworzona do celów rozrywkowych, choć na naszej liście znajdują się również aplikacjie praktyczne pomocne w edukacji czy w robieniu zakupów.*

In [17]:
#Analiza tebeli częstości wygenerowanej dla Category:
display_table(android_final, 1)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

*Najpopularniejszymi gatunkami są: aplikacje rodzinne (gry dla dzieci), gry i narzędzia. Na rynku Google Play jest dużo mniej gire (tylko 9,7% ) niż na rynku Apple Store (58,2%). Choć tak duża różnica może wynikać z faktu, iż Google Play posiada osobną kategorię dla gire- family, Apple nie posiada takiego podziału. Dla obu rynków zdecydowanie najbardziej korzystnymi aplikacjami są gry.*

In [18]:
#Obliczam średnią liczbę ocen użytkowników według gatunku aplikacji w App Store:
genres_ios = freq_table(ios_final, -5)

for genre in genres_ios:
    total = 0
    len_genre = 0
    for app in ios_final:
        genre_app = app[-5]
        if genre_app == genre:            
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
    avg_n_ratings = total / len_genre
    print(genre, ':', avg_n_ratings)

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


*Najwięcej opini mają aplikacje nawigcyjne choć wcale nie musi to oznaczać, że są to najbardziej opłacalne aplikacje. Może to wynkać z błędów w aplikacji, które prowadzą użytkowników po nieporządanych drogach (komu się to nigdy nie przydarzyło? :))*


In [19]:
#Średnia liczba ocen użytkowników:
for app in ios_final:
    if app[-5] == 'Navigation':
        print(app[1], ':', app[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


In [20]:
#Gatunek aplikacji i średnia liczba ocen użytkowników:
for app in ios_final:
    if app[-5] == 'Reference':
        print(app[1], ':', app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


*Aplikacje referencyjne mają średnio 74 942 oceny użytkowników, ale w rzeczywistości to Biblia  i Dictionary.com podnoszą średnią ocenę: Wydaje się jednak, że ta nisza ma pewien potencjał. Jedną z rzeczy, które moglibyśmy zrobić, to wziąć inną popularną książkę i przekształcić ją w aplikację, w której moglibyśmy dodać różne funkcje oprócz surowej wersji książki. Może to obejmować codzienne cytaty z książki, wersję audio książki, quizy dotyczące książki itp. Co więcej, moglibyśmy również osadzić słownik w aplikacji, dzięki czemu użytkownicy nie musieliby opuszczać naszej aplikacji, aby wyszukać słowa w aplikacji zewnętrznej.*

*Ten pomysł wydaje się dobrze pasować do faktu, że App Store jest zdominowany przez aplikacje do zabawy. Sugeruje to, że rynek może być nieco przesycony trgo rodzaju aplikacjami, co oznacza, że praktyczna aplikacja może mieć większą szansę na wyróżnienie się wśród ogromnej liczby aplikacji w App Store.*


In [21]:
#Tabela częstości dla kolumny Category zbioru danych Google Play:
display_table(android_final, 5)

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


*Mamy dane na temat liczby instalacji na rynku Google Play, więc powinniśmy uzyskać jaśniejszy obraz popularności gatunku. Jednak liczby instalacji nie wydają się wystarczająco dokładne — widzimy, że większość wartości jest nieograniczona (100+, 1000+, 5000+). Nie potrzebujemy jednak do naszych celów bardzo dokładnych danych — chcemy jedynie dowiedzieć się, które gatunki aplikacji przyciągają najwięcej użytkowników.*



In [22]:
# Przeglądam unikalne gatunki ze zbiorów danych Google Play:
categories_android = freq_table(android_final, 1)

for category in categories_android:
    total = 0
    len_category = 0
    for app in android_final:
        category_app = app[1]
        if category_app == category:            
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
            
# Oblicznie średniej liczby instalacji:
    avg_n_installs = total / len_category
    print(category, ':', avg_n_installs)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

Dane dotyczące średnich instalacji wskazują na kilka trendów:

1. **Najpopularniejsze:**
   - Komunikacja i Odtwarzacze Wideo mają najwyższą średnią liczbę instalacji.

2. **Popularne kategorie:**
   - Rozrywka, Produktywność, Gry i Aplikacje Rodzinne przyciągają dużo uwagi.

3. **Interesujące, ale niszowe:**
   - Aplikacje związane z Pogodą, Wiadomościami oraz Zakupami cieszą się zainteresowaniem.

4. **Mniejsze zainteresowanie:**
   - Kategorie takie jak Wydarzenia, Randki czy Rodzicielstwo mają niższą średnią liczbę instalacji.



In [23]:
# Obliczanie instalacji wśród aplikacji komunikacyjnych:
for app in android_final:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

*Aplikacje związane z komunikacją wydają się być najbardziej popularne, prezentując średnią liczbę instalacji wynoszącą aż 38,456,119. Jednakże warto zauważyć, że ta liczba jest znacząco zniekształcona przez kilka gigantycznych aplikacji, takich jak WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail i Hangouts, które osiągają ponad miliard instalacji. Ponadto, istnieje kilka innych aplikacji z liczbą instalacji przekraczającą 100 i 500 milionów, co także znacząco wpływa na średnią. To sugeruje, że choć ogólnie kategoria komunikacyjna cieszy się ogromnym zainteresowaniem, to głównie kilka liderów rynku przyczynia się do tej imponującej średniej liczby instalacji.*

In [24]:
# Przeglądanie 
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

*Kategoria książek i materiałów referencyjnych obejmuje zróżnicowane zastosowania, takie jak oprogramowanie do przetwarzania e-booków, różnorodne kolekcje bibliotek, słowniki oraz samouczki dotyczące programowania lub języków programowania. Wciąż istnieje niewielka liczba aplikacji, które cieszą się ogromną popularnością.*

In [25]:
# Szukanie aplikacji znajdujących się pośrodku pod względem popularności:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

*W tej dziedzinie wydaje się, że dominują aplikacje do przetwarzania i czytania e-booków, a także różnorodne zasoby bibliotek i słowników. Tworzenie podobnych aplikacji może być ryzykowne ze względu na silną konkurencję.*

*Zauważyłam także znaczną obecność aplikacji opartych na konkretnych tekstach, na przykład Koranie. To sugeruje, że adaptacja popularnej książki, zwłaszcza nowszej, na aplikację może być rentownym przedsięwzięciem zarówno na platformie Google Play, jak i App Store.*

*Niemniej jednak, patrząc na obecność licznych bibliotek, aby wyróżnić się, musimy dostarczyć coś więcej niż surową wersję książki. Możemy rozważyć dodanie dodatkowych funkcji, takich jak codzienne cytaty z książki, wersja audio, quizy związane z treścią, forum do dyskusji i inne innowacyjne elementy, które przyciągną użytkowników.*

**Conclusions**


W trakcie projektu przeprowadziłam analizę danych dotyczących aplikacji mobilnych dostępnych na App Store i Google Play, aby opracować rekomendacje dotyczące opłacalnych profili aplikacji na obu rynkach.

Analizy wskazują, że adaptacja popularnej książki, zwłaszcza nowszej, na formę aplikacji może przynieść korzyści zarówno na platformie Google Play, jak i App Store. Zważywszy na obfitość istniejących bibliotek, uznałam jednak, że kluczowe jest wprowadzenie dodatkowych funkcji, aby wyróżnić się spośród konkurencji. Potencjalne elementy takie jak codzienne cytaty z książki, wersja audio, quizy dotyczące treści, a także forum do dyskusji, mają potencjał przyciągnięcia użytkowników i zwiększenia atrakcyjności aplikacji.
